### ISSUE TITLE
Duty cycle is halved using LEDC PWM on ESP32-C3

### ISSUE THREAD

**Issue by DurandA** [wokwi/wokwi-features#379]
**Describe the bug**
PWM signals generated by the [LEDC peripheral](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/ledc.html) have half of the expected duty cycle on the ESP32-C3 (and not on ESP32).

**To Reproduce**
[Simulation project](https://wokwi.com/projects/336380083260359252) with logic analyzer using the [official LEDC example](https://github.com/espressif/esp-idf/blob/release/v5.0/examples/peripherals/ledc/ledc_basic/main/ledc_basic_example_main.c).

**Expected behavior**
GPIO5/D0 should have a duty cycle of 50% instead of 25% in the above example.

**Environment (please complete the following information):**
 - OS: Linux Mint 20.3
 - Browser: Firefox 101

**Comment by Dlloydev** [comment]
Looks to me like the bug is in the example itself which predates their latest release [Arduino ESP32 Version 2.0.3](https://github.com/espressif/arduino-esp32/releases) which has fixes on LEDC.

Their [LEDC ESP32-C3 reference](https://docs.espressif.com/projects/esp-idf/en/latest/esp32c3/api-reference/peripherals/ledc.html) states:

> The range of the duty cycle values passed to functions depends on selected duty_resolution and should be from 0 to (2 ** duty_resolution) - 1. For example, if the selected duty resolution is 10, then the duty cycle values can range from 0 to 1023. This provides the resolution of ~0.1%.

I have a library that works with LEDC PWM [here](https://github.com/Dlloydev/ESP32-ESP32S2-AnalogWrite) and has the following examples where the duty is as expected:
Wokwi phase shift examples ... [ESP32](https://wokwi.com/projects/334722465700774482) ... [ESP32-S2](https://wokwi.com/projects/334765722024542804) ... [ESP32-C3](https://wokwi.com/projects/334856585002091092)

Note that in [pwmWrite.cpp](url), I set the bit resolution as follows:

```c++
#if (CONFIG_IDF_TARGET_ESP32S2 || CONFIG_IDF_TARGET_ESP32S3 || CONFIG_IDF_TARGET_ESP32C3)
      bits = ((bits - 1) & 0xF);
#endif
```

**Comment by DurandA** [comment]
> Looks to me like the bug is in the example itself which predates their latest release [Arduino ESP32 Version 2.0.3](https://github.com/espressif/arduino-esp32/releases) which has fixes on LEDC.

If you try the exact same Wokwi project with an ESP32 instead of an ESP32-S3, then the measured duty cycle is 50% as expected.

**Comment by urish** [comment]
Thanks for the super detailed bug report!

I can confirm this issue reproduces. This is the output I see in the simulator:

![image](https://user-images.githubusercontent.com/892318/177484283-4c3cdf8b-954e-4944-b462-ead58173c1c2.png)

And this is the output I can see running the same binary (obtained by pressing "F1" -> "Download Compiled HEX Firmware") on a physical device:

![image](https://user-images.githubusercontent.com/892318/177483799-5b620684-6de5-4c22-8b4b-fcb3050bffe4.png)

To summarize: In the simulator, the signal goes high for 100 us, low for 300 us. Where as on the physical device, it goes high for 100 us, low for 100 us.

**Comment by urish** [comment]
Uploaded a fix - can you please try again now and report?

**Comment by DurandA** [comment]
The fix works fine.

Thank you for your awesome project.
